File: C Code/Basics/Array/1 D/Insertion Sort.c
#include<stdio.h>
int main(){
  int n;
  printf("Enter the size of array : ");
  scanf("%d",&n);
  int arr[n];
  printf("Enter elements :\n");
  for(int i=0;i<n;i++)
    scanf("%d",&arr[i]);
  for(int i=1;i<n;i++){
    int temp=arr[i],j=i-1;
    while(j>=0){
      if(arr[j]>temp){
        arr[j+1]=arr[j];
        j--;
      }
      else
        break;
    }
    arr[j+1]=temp;
  }
  printf("Sorted Array :\n");
  for(int i=0;i<n;i++)
    printf("%d ",arr[i]);
  return 0;
}

File: Javascript Code/Array/InsertionSort.js
const readline = require('readline');

const r1 = readline.Interface(process.stdin,process.stdout);

function Select(arr,k){
	let ans=k;
	for(let i=k+1;i<arr.length;i++){
		if(arr[i]<arr[ans])
			ans=i;
	}
	return ans;
}
function SelectionSort(arr,n){
	if(n<2)
		return ;
	for(let i=0;i<n;i++){
		let idx=Select(arr,i);
		let temp=arr[i];
		arr[i]=arr[idx];
		arr[idx]=temp;
	}
}

r1.question("Enter all the elements of the array : ",(input)=>{
  const arr=input.split(' ').map(Number);
	SelectionSort(arr,arr.length);
	console.log(`Sorted array : ${arr}`);
	r1.close();
});